plugins/modules/mattermost.py

```diff
@@ -62,13 +62,17 @@
   username:
     type: str
     description:
-      - This is the sender of the message (Username Override need to be enabled by mattermost admin, see mattermost doc.
+      - This is the sender of the message (Username Override need to be enabled by mattermost admin, see mattermost doc).
     default: Ansible
   icon_url:
     type: str
     description:
       - URL for the message sender's icon.
     default: https://docs.ansible.com/favicon.ico
+  priority:
+    type: str
+    description:
+      - Set a priority for the message. For supported values, see mattermost doc.
```

Comment:
I would strongly recommend listing the supported values as `choices` here, so that the module does not need to send a value all the way to the API and back when it is invalid.

Reply:
I have thought of that. But those values need to be changed if the server changes them. Wouldn't it be more flexible to just let the server decide what it accepts?

Reply:
Testing the `choices` change now and will commit soon.
Will also add `priority: important` to one of the examples and remove the second sentence from `priority` documentation.

Reply:
@bluikko your questioning is 100% valid, and the only good answer to that would be: how often do they change the values in the API? Or, how often do they break backwards compatibility in the API? I would expect this to be a very rare event, in which case we would save time for a lot of people by checking the value sooner rather than later.